Implement more smart pointers in WebProcess.h/cpp (part 4)

lopezlozoya opened the pull request:
#### d0e0bfc2ddc9179626131382ba149190bdf9f233
<pre>
Implement more smart pointers in WebProcess.h/cpp (part 4)
<a href="https://bugs.webkit.org/show_bug.cgi?id=288252">https://bugs.webkit.org/show_bug.cgi?id=288252</a>

Reviewed by Ryosuke Niwa.

Deploy more smart pointer fixes and downcasting fixes (part 4).

* Source/WebCore/workers/service/context/SWContextManager.h:
(WebCore::SWContextManager::Connection::isWebSWContextManagerConnection const):
* Source/WebCore/workers/shared/context/SharedWorkerContextManager.h:
(WebCore::SharedWorkerContextManager::Connection::isWebSharedWorkerContextManagerConnection const):
* Source/WebKit/SaferCPPExpectations/MemoryUnsafeCastCheckerExpectations:
* Source/WebKit/WebProcess/Storage/WebSWContextManagerConnection.h:
(isType):
* Source/WebKit/WebProcess/Storage/WebSharedWorkerContextManagerConnection.h:
(isType):
* Source/WebKit/WebProcess/WebPage/WebFrame.h:
(isType):
* Source/WebKit/WebProcess/WebPage/WebPage.h:
(isType):
* Source/WebKit/WebProcess/WebProcess.cpp:
(WebKit::WebProcess::dispatchMessage):
(WebKit::WebProcess::transformHandlesToObjects):
(WebKit::WebProcess::transformObjectsToHandles):

Canonical link: <a href="https://commits.webkit.org/291102@main">https://commits.webkit.org/291102@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0806e4955312661882f845e3aedebb003301ea05

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/91965 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/11496 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/1042 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/96897 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/42566 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/94015 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/11839 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/19982 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/70561 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/28043 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/94966 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/9020 "Found 1 new test failure: fast/forms/ios/focus-input-in-fixed.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/83295 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/50889 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/8749 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/894 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/41781 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/79071 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/890 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/98929 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/19083 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/14105 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/79592 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/19335 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/79145 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/78820 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/23351 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/669 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/12121 "The change is no longer eligible for processing.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/14611 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/19064 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/24273 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/18761 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/22220 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/20512 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WebCore/workers/service/context/SWContextManager.h:
```diff
@@ -92,6 +92,8 @@ class SWContextManager {
 
         virtual void removeNavigationFetch(SWServerConnectionIdentifier, FetchIdentifier) = 0;
 
+        virtual bool isContextManagerConnection() const { return true; }
```
We should be returning false here, right?

geoffreygaren replied:
Possibly also rename this to `isWebSWContextManagerConnection()`, since `WebSWContextManagerConnection` is the subclass name.

lopezlozoya replied:
Got it, I'll fix this and add the method on the subclass.

geoffreygaren commented on Source/WebCore/workers/shared/context/SharedWorkerContextManager.h:
```diff
@@ -55,6 +55,7 @@ class SharedWorkerContextManager {
         virtual void postErrorToWorkerObject(SharedWorkerIdentifier, const String& errorMessage, int lineNumber, int columnNumber, const String& sourceURL, bool isErrrorEvent) = 0;
         virtual void sharedWorkerTerminated(SharedWorkerIdentifier) = 0;
         bool isClosed() const { return m_isClosed; }
+        virtual bool isContextManagerConnection() const { return true; }
```
This fella wants to return false too.

Possibly also rename this to `isWebSharedWorkerContextManagerConnection`.

geoffreygaren reviewed (changes requested):
Looks good but we need to fix the virtual functions because they currently always permit the cast, even when it's invalid.

lopezlozoya committed:
Implement more smart pointers in WebProcess.h/cpp (part 4)
https://bugs.webkit.org/show_bug.cgi?id=288252

Reviewed by Ryosuke Niwa.

Deploy more smart pointer fixes and downcasting fixes (part 4).

* Source/WebCore/workers/service/context/SWContextManager.h:
(WebCore::SWContextManager::Connection::isWebSWContextManagerConnection const):
* Source/WebCore/workers/shared/context/SharedWorkerContextManager.h:
(WebCore::SharedWorkerContextManager::Connection::isWebSharedWorkerContextManagerConnection const):
* Source/WebKit/SaferCPPExpectations/MemoryUnsafeCastCheckerExpectations:
* Source/WebKit/WebProcess/Storage/WebSWContextManagerConnection.h:
(isType):
* Source/WebKit/WebProcess/Storage/WebSharedWorkerContextManagerConnection.h:
(isType):
* Source/WebKit/WebProcess/WebPage/WebFrame.h:
(isType):
* Source/WebKit/WebProcess/WebPage/WebPage.h:
(isType):
* Source/WebKit/WebProcess/WebProcess.cpp:
(WebKit::WebProcess::dispatchMessage):
(WebKit::WebProcess::transformHandlesToObjects):
(WebKit::WebProcess::transformObjectsToHandles):

Canonical link: https://commits.webkit.org/291102@main
